components/object_store/src/mem_cache.rs

```diff
@@ -209,6 +230,10 @@ impl ObjectStore for MemCacheStore {
         // TODO(chenxiang): What if two threads reach here? It's better to
         // pend one thread, and only let one to fetch data from underlying store.
         let bytes = self.underlying_store.get_range(location, range).await;
+        if self.read_only_cache {
```

Comment:
I think of one issue with current implementation.

Although this read only cache will ensure mem isn't wasted, it will make LRU useless, it may better to call `pop` method for this case.
- https://docs.rs/clru/0.6.1/clru/struct.CLruCache.html#method.peek

It will not update LRU position.

Reply:
👌